src/actions/index.js

```diff
@@ -1,8 +1,16 @@
-// we'll need axios
+import axios from 'axios';
 
-// we'll need to create 3 different action types here.
-// one for fetching, one for success and one for failure
+export const FETCH_CHARACTERS_START = "FETCH_CHARACTERS_START";
+export const FETCH_CHARACTERS_SUCCESS = "FETCH_CHARACTERS_SUCCESS";
+export const FETCH_CHARACTERS_FAILURE = "FETCH_CHARACTERS_FAILURE";
 
-// our action creator will be a function that returns a function
-// the url to fetch characters from is `https://swapi.co/api/people/`
-// remember that now we have controll over our thunk-based action creator
+
+export const getCharacters = () => dispatch => {
+    dispatch({ type: FETCH_CHARACTERS_START });
+    axios
+      .get(
+        "https://swapi.co/api/people/"
+      )
+      .then(res => dispatch({ type: FETCH_CHARACTERS_SUCCESS, payload: res.data.results }))
```

Comment:
great use of dispatch here, your error dispatch can also be good if you actually want to collect the error and do something with it which might not always be the case. Sometimes you want to just let the error go out quietly in places that might not disturb your site being used.